src/Microsoft.ServiceFabric.Services.Queryable.Test/QueryPartitionTests.cs

```diff
@@ -0,0 +1,426 @@
+            string collection = "collection";
+            IEnumerable<KeyValuePair<string, string>> query = new List<KeyValuePair<string, string>>();
+
+            var stateManager = new Mock<IReliableStateManager>();
+            stateManager.Setup(obj => obj.TryGetAsync<IReliableState>(collection)).Returns(Task.FromResult(new ConditionalValue<IReliableState>(true, queue.Object)));
+
+
+            var result = await ReliableStateExtensions.QueryPartitionAsync(stateManager.Object, httpContext.Object, collection, query, Guid.NewGuid(), new CancellationToken());
+        }
+
+        private HashSet<string> getProfilesFromJTokens(IEnumerable<JToken> result)
+        {
+            HashSet<string> profiles = new HashSet<string>();
+            foreach (JToken token in result)
+            {
+                profiles.Add(token.First.Next.Next.ToString());
+            }
+            return profiles;
+        }
+
+        [TestMethod]
+        public async Task EmptyQuery_NotIndexed_ReturnsUserProfile1()
+        {
+            var httpContext = new Mock<HttpContext>();
+            httpContext.Setup(obj => obj.TraceIdentifier).Returns("Test trace");
+            IEnumerable<KeyValuePair<string, string>> query = new List<KeyValuePair<string, string>>();
+
+            IEnumerable<JToken> result = await ReliableStateExtensions.QueryPartitionAsync(userDictionaryManager, httpContext.Object, "users", query, Guid.NewGuid(), new CancellationToken());
+            var profiles = getProfilesFromJTokens(result);
+            Assert.IsTrue(profiles.Contains(user0));
+            Assert.IsTrue(profiles.Contains(user1));
+            Assert.IsTrue(profiles.Contains(user2));
+            Assert.IsTrue(profiles.Contains(user3));
+            Assert.IsTrue(profiles.Contains(user4));
+        }
+
+        [TestMethod]
+        public async Task Top1Query_NotIndexed_Returns1UserProfile()
+        {
+            var httpContext = new Mock<HttpContext>();
+            httpContext.Setup(obj => obj.TraceIdentifier).Returns("Test trace");
+            List<KeyValuePair<string, string>> query = new List<KeyValuePair<string, string>>();
+            query.Add(new KeyValuePair<string, string>("$top", "1"));
+
+            IEnumerable<JToken> result = await ReliableStateExtensions.QueryPartitionAsync(userDictionaryManager, httpContext.Object, "users", query, Guid.NewGuid(), new CancellationToken());
+            var profiles = getProfilesFromJTokens(result);
+            int profilesreturned = 0;
+
+            if (profiles.Contains(user0))
+                profilesreturned++;
+            if (profiles.Contains(user1))
+                profilesreturned++;
+            if (profiles.Contains(user2))
+                profilesreturned++;
+            if (profiles.Contains(user3))
+                profilesreturned++;
+            if (profiles.Contains(user4))
+                profilesreturned++;
+
+            Assert.AreEqual(1, profilesreturned);
```

Comment:
I would assert that the length of `profiles` is 1 and that the only entry is in `new [] { user0, user1, user2, user3, user4 }`